scripts/docs-deploy/deploy-ci-push.ts

```diff
@@ -0,0 +1,112 @@
+#!/usr/bin/env node
+
+import {
+  fetchActiveReleaseTrains,
+  getBranchesForMajorVersions,
+  getVersionForVersionBranch,
+  isVersionBranch,
+} from '@angular/dev-infra-private/ng-dev';
+import * as semver from 'semver';
+import {getReleaseRepoWithApi} from './github-versioning';
+import {buildAndDeployWithSnapshots} from './snapshot-deploy';
+import {updateVersionsFile} from './update-versions-file';
+import {firebaseConfig, getPackageJsonOfProject, projectDir, sites} from './utils';
+
+async function main() {
+  if (process.env.CIRCLE_PR_NUMBER !== undefined) {
+    console.log('Skipping deployment for pull request build.');
+    return;
+  }
+
+  const branchName = process.env.CIRCLE_BRANCH;
+  if (branchName === undefined) {
+    throw new Error('Deployment script is unable to determine CI branch.');
+  }
+
+  const repo = getReleaseRepoWithApi();
+  const active = await fetchActiveReleaseTrains(repo);
+  const description = `${branchName} - ${process.env.CIRCLE_SHA1!}`;
+  const {projectId, token: authToken} = firebaseConfig;
+
+  if (branchName === active.next.branchName) {
+    const major = active.next.version.major;
+    const targets = [{projectId, description, site: sites.next}];
+
+    // If the next release train is for a new major that is not published as part of the
+    // other active release trains, we also publish to e.g. `v14.material.angular.io`.
+    if (major > (active.releaseCandidate ?? active.latest).version.major) {
+      targets.push({projectId, description, site: sites.forMajor(major)});
+    }
+
+    await buildAndDeployWithSnapshots(authToken, major, targets);
+    return;
+  }
+
+  if (branchName === active.latest.branchName) {
+    const major = active.latest.version.major;
+    const targets = [
+      {projectId, description, site: sites.stable},
+      {projectId, description, site: sites.forMajor(major)},
+    ];
+
+    // If there is no active RC train, we also push the current stable to the `rc` site.
+    // TODO: This can be improved by using redirects from rc -> stable.
+    if (active.releaseCandidate === null) {
+      targets.push({projectId, description, site: sites.rc});
+    }
+
+    await buildAndDeployWithSnapshots(authToken, major, targets, {
+      prebuild: docsRepoDir => updateVersionsFile(docsRepoDir, active),
```

Comment:
OOC, why is the versions file only updated when deploying the stable branch?

Reply:
I will add a comment. All other doc sites consult the `versions.json` file from `https://material.angular.io` -- obviously will not work properly for my testing firebase project but that is acceptable.